designcompose/src/main/java/com/android/designcompose/squoosh/SquooshAnimate.kt

```diff
@@ -74,11 +93,30 @@ internal class SquooshAnimatedLayout(
             return
         }
 
+        val fromWidth = if (hasRotation) from.view.style.fixedWidth(density) else fromLayout.width
+        val fromHeight =
+            if (hasRotation) from.view.style.fixedHeight(density) else fromLayout.height
+        val toWidth = if (hasRotation) to.view.style.fixedWidth(density) else toLayout.width
+        val toHeight = if (hasRotation) to.view.style.fixedHeight(density) else toLayout.height
+
+        if (hasRotation) {
+            // Interpolate the transformation matrix
+            val transform: MutableList<Float> = MutableList(16) { 0F }
+            for (i in 0..15) {
+                transform[i] = toTransform[i] * value + fromTransform[i] * iv
+            }
+            target.style = target.style.withNodeStyle { s -> s.transform = Optional.of(transform) }
+        }
+
+        // When doing a layout animation, set overrideLayoutSize to true so that the rendering code
+        // uses this computed size as opposed to the size of the node in ViewStyle. This ensures
+        // that size change animations and rotation animations render at the correct size.
+        target.overrideLayoutSize = true
```

Comment:
Wouldn't this make a discontinuity at the start of an animation where we need to use the style size (instead of the layout size)?

I really dislike how Figma's rotation implementation causes so much complexity in our runtime, so I am wondering if there's another way of handling it during doc conversion so that the runtime doesn't have to deal with it (or if there's a different way to express it in the runtime?). We shouldn't hold up this change on a new design for rotation, tho. 

PTAL at any discontinuous cases, we should fix those because they'll definitely come up in partner designs.

Reply:
There are (at least) two scenarios to consider here:
1. Without rotations: without any rotations, the code here uses the computed layout sizes. Setting `overrideLayoutSize` to true forces the renderer to use this size instead of the Figma node size, which is what it was doing before anyway because you had changed the `rectSize` parameter to always use the layout size.
2. With rotations: Instead of the layout size, we use the raw size from Figma. This means that if layout changed that size, then yes there would be a discontinuity. However I don't think we previously supported rotated nodes with contraints that would stretch or shrink them, and we still don't, so I thought that would be ok here.

Reply:
Oh, right, I had forgotten the "rotated with constraints layout" case :(. We don't support it, and Figma is doing so much UI stuff it's a little hard to figure out exactly what's happening behind the scenes (or how to make another layout engine support the same concept -- CSS does rotation strictly as a non-layout changing render effect, which is conceptually a lot easier to understand and implement, but perhaps a bit harder for content authoring).

Makes sense to keep it as you have it...